internal/backends/wasm/wasm.go

```diff
@@ -0,0 +1,221 @@
+                        ...on PackageVersion {
+                            package {
+                                name
+                            }
+                            version
+                            description
+                            homepage
+                            repository
+                            license
+                            publishedBy {
+                                username
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        `)
+		// set any variables
+		req.Var("query", string(query))
+		ctx := context.Background()
+
+		var search searchQuery
+		if err := client.Run(ctx, req, &search); err != nil {
+			util.Die("WAPM response: %s", err)
+		}
+
+		results := []api.PkgInfo{}
+		for _, edge := range search.Search.Edges {
+			results = append(results, api.PkgInfo{
+				Name:        edge.Node.Package.Name,
+				Description: edge.Node.Description,
+				Version:     edge.Node.Version,
+				HomepageURL: edge.Node.Homepage,
+				Author:      edge.Node.PublishedBy.Username,
+				License:     edge.Node.License,
+			})
+		}
+		return results
+	},
+	Info: func(name api.PkgName) api.PkgInfo {
+		client := graphql.NewClient("https://registry.wapm.io/graphql")
+		req := graphql.NewRequest(`
+        query getPackage($name: String!) {
+            packageVersion: getPackageVersion(name: $name) {
+                package {
+                    name
+                }
+                version
+                description
+                homepage
+                repository
+                license
+                publishedBy {
+                    username
+                }
+            }
+        }
+        `)
+		// set any variables
+		req.Var("name", string(name))
+		ctx := context.Background()
+
+		var output infoQuery
+		if err := client.Run(ctx, req, &output); err != nil {
+			util.Die("WAPM response: %s", err)
+		}
+
+		return api.PkgInfo{
+			Name:        output.PackageVersion.Package.Name,
+			Description: output.PackageVersion.Description,
+			Version:     output.PackageVersion.Version,
+			HomepageURL: output.PackageVersion.Homepage,
+			Author:      output.PackageVersion.PublishedBy.Username,
+			License:     output.PackageVersion.License,
+		}
+	},
+	Add: func(pkgs map[api.PkgName]api.PkgSpec) {
+		if !util.Exists("wapm.lock") {
+			util.RunCmd([]string{"wapm", "init", "-y"})
+		}
+		cmd := []string{"wapm", "add"}
+		for name, spec := range pkgs {
+			arg := string(name)
+			if spec != "" {
+				arg += "@" + string(spec)
+			}
+			cmd = append(cmd, arg)
+		}
+		util.RunCmd(cmd)
+	},
+	Remove: func(pkgs map[api.PkgName]bool) {
+		cmd := []string{"wapm", "remove"}
+		for name := range pkgs {
+			cmd = append(cmd, string(name))
+		}
+		util.RunCmd(cmd)
+	},
+	Lock: func() {
+		util.RunCmd([]string{"wapm", "install"})
+	},
+	Install: func() {
+		util.RunCmd([]string{"wapm", "install", "-y"})
```

Comment:
I noticed Lock uses `wapm install` and Install uses `wapm install -y`. What's the difference? I don't see it mentioned in the docs.

Reply:
`wapm install -y` automatically accepts certificates for signed packages when installing them (and any other input requested from the user).

Reply:
sounds like we want `wapm install -y` in both cases then

Reply:
@turbio sounds good! I just updated the PR :)